File: src/Sales/BrewUp.Sales.Facade/SalesFacadeHelper.cs
using BrewUp.Sales.Domain;
using BrewUp.Sales.Facade.Acl;
using BrewUp.Sales.Infrastructure;
using BrewUp.Sales.ReadModel;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Muflone;

namespace BrewUp.Sales.Facade;

public static class SalesFacadeHelper
{
    public static IServiceCollection AddSalesFacade(this IServiceCollection services,
        IConfigurationManager configurationManager)
    {
        services.AddScoped<ISalesFacade, SalesFacade>();

        services.AddSalesDomain();
        services.AddSalesReadModel();
        services.AddSalesInfrastructure(configurationManager);

        services.AddIntegrationEventHandler<SalesOrderProductsPreparedEventHandler>();

        return services;
    }
}

File: src/Sales.FSharp/BrewUp.Sales.Facade.FSharp/SalesFacadeHelper.fs
namespace BrewUp.Sales.Facade.FSharp

open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Configuration
open BrewUp.Sales.Domain.FSharp

module SalesFacadeHelper =
    
    let addSalesFacade (services: IServiceCollection) (configuration: IConfiguration) =
        services.AddScoped<ISalesFacade, SalesFacade>() |> ignore
        
        SalesDomainHelper.addSalesDomain services |> ignore
        
        services